pull_request_template.md

```diff
@@ -0,0 +1,35 @@
+# PULL REQUEST
+
+<!--
+Please read and fill out this form before submitting your PR.
+
+Please make sure you have reviewed our contributors guide before submitting your
+first PR.
+-->
+
+## Overview
+
+<!-- 
+Please provide an explanation of the PR, including the apprioprate context,
+background, goal, and rationale. If there is an issues with this information,
+please provide a tl;dr and link the issue. 
+-->
+
+## Checklist
+
+<!-- 
+Review and complete the checklist to ensure that the PR is complete before
+assigned to an approver.  
+
+IMPORTANT:
+PRs should be left in Draft until the below checklist is completed.
+-->
+
+- [ ] New and updated code has appropriate documentation
+- [ ] New and update code has new and/or updated testing
+- [ ] PR is linked to appropriate project
+- [ ] CI is passing
+- [ ] Reviewers have been properly assigned
+- [ ] PR is self assigned
```

Comment:
[discussion] I think the value of this checklist diminishes if we have tedious items to fill out. In other words, I expect populating this checklist on all PRs will become a chore for repo maintainers. 

Similar to issue templates, I think a PR template may be helpful for external contributors but they don't have permission to request reviewers on PRs so this item isn't helpful for them. 

Is the target audience of this PR template for all PRs, external contributor authored PRs, or subject to sub-team discretion?

If external contributors:

```suggestion
```

Reply:
The idea was that this is a base PR template to cover all repos. But individual repos can always overwrite it by making a PR template in their repo.

Fine to remove these, these should be automated actions anyways.